tests/unit/test_state.py

```diff
@@ -610,3 +611,70 @@ def test_builder_app_channel_from_charm_error():
         state.BuilderAppChannel.from_charm(charm=charm)
 
     assert "invalid" in str(exc.getrepr())
+
+
+@pytest.mark.parametrize(
+    "exception, expected_error",
+    [
+        pytest.param(ops.SecretNotFoundError, "Secret label not found:"),
+        pytest.param(ops.ModelError, "Please grant the charm read access to the secret."),
+    ],
+)
+def test__parse_script_secrets_invalid_secret(exception: Exception, expected_error: str):
+    """
+    arrange: given a mocked model get_secret method that raises a given error.
+    act: when _parse_script_secrets is called.
+    assert: InvalidSecretError is raised.
+    """
+    mock_charm = MagicMock()
+    mock_charm.config = {state.SCRIPT_SECRET_ID_CONFIG_NAME: "secret-label"}
+    mock_charm.model = MagicMock()
+    mock_charm.model.get_secret = MagicMock(side_effect=exception)
+
+    with pytest.raises(state.InvalidSecretError) as exc:
+        state._parse_script_secrets(charm=mock_charm)
+
+    assert expected_error in str(exc)
+
+
+def test__parse_script_secrets_invalid_key_value_pair():
```

Comment:
Should we parametrise this with a few possible invalid values? E.g., empty string, one config with no = multiple value with both no =, one of them no =